src/main/java/com/example/notes/mapper/NoteMapper.java

```diff
@@ -0,0 +1,21 @@
+package com.example.notes.mapper;
+
+import com.example.notes.dto.NoteDto;
+import com.example.notes.entity.NoteEntity;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Component
+public class NoteMapper {
+
+    public List<NoteDto> mapEntityToDto(List<NoteEntity> noteEntityList) {
+        return noteEntityList.stream()
+                .map(noteEntity -> new NoteDto(
+                        noteEntity.getContent(),
+                        noteEntity.getCreatedAt()))
+                .collect(Collectors.toList());
+    }
+
+}
```

Comment:
It is much better for this component to map only 1 entity to dto (1 thing at a time). The mapper must only map 1=>1 thing. It would be easier to test and manage in the future.

It is then possible to map a list in the service (or we can add a 2nd public method here in the mapper)

Reply:
Got it!